test/e2e/certman_operator_tests.go

```diff
@@ -62,4 +80,186 @@ var _ = Describe("Certman Operator", Ordered, func() {
+			if err != nil {
+				logger.Info("CertificateRequest is deleted", "name", crName)
+				return true
+			}
+
+			finalizers := updatedCR.GetFinalizers()
+			for _, finalizer := range finalizers {
+				if finalizer == "certificaterequests.certman.managed.openshift.io" {
+					logger.Info("Finalizer has been re-added to CertificateRequest", "name", crName)
+					return true
+				}
+			}
+
+			logger.Info("Finalizer not yet re-added to CertificateRequest", "name", crName)
+			return false
+
+		}, pollingDuration, 30*time.Second).Should(BeTrue(), "CertificateRequest should be deleted or have finalizer re-added")
+	})
+
+	It("should have ClusterDeployment as the owner of the CertificateRequest", func(ctx context.Context) {
+
+		clusterDeploymentGVR := schema.GroupVersionResource{
+			Group:    "hive.openshift.io", 
+			Version:  "v1",
+			Resource: "clusterdeployments",
+		}
+
+		logger.Info("Checking if ClusterDeployment exists.")
+
+		clusterDeploymentList, err := dynamicClient.Resource(clusterDeploymentGVR).Namespace("certman-operator").List(ctx, metav1.ListOptions{})
+		if err != nil {
+			Fail("Error fetching ClusterDeployments")
+		}
+
+		if len(clusterDeploymentList.Items) == 0 {
+			Fail("ClusterDeployment not found.")
+		}
+
+		logger.Info("ClusterDeployment found, proceeding to check Owner Reference")
+
+		clusterDeploymentGVR = schema.GroupVersionResource{
+			Group:    "certman.managed.openshift.io",
+			Version:  "v1alpha1",
+			Resource: "certificaterequests",
+		}
+
+		logger.Info("Fetching CertificateRequests...")
+
+		Eventually(func() bool {
+			crList, err := dynamicClient.Resource(clusterDeploymentGVR).Namespace("certman-operator").List(ctx, metav1.ListOptions{})
+
+			if err != nil || len(crList.Items) == 0 {
+				logger.Error(err, "Error fetching CertificateRequests")
+				return false
+			}
+
+			certRequest := crList.Items[0]
+			logger.Info("Found CertificateRequest", "name", certRequest.GetName())
+
+			ownerRefs := certRequest.GetOwnerReferences()
+			logger.Info("Found OwnerReferences", "ownerRefs", ownerRefs)
+
+			var clusterDeploymentOwnerFound bool
+			for _, owner := range ownerRefs {
+				logger.Info("Checking owner", "kind", owner.Kind, "name", owner.Name)
+				if owner.Kind == "ClusterDeployment" && owner.Name == clusterName {
+					logger.Info("Found ClusterDeployment as owner!")
+					clusterDeploymentOwnerFound = true
+					break
+				}
+			}
+
+			if !clusterDeploymentOwnerFound {
+				logger.Info("ClusterDeployment is not the owner, adding it as the owner...")
+
+				isTrue := true
+				ownerRef := metav1.OwnerReference{
+					APIVersion:         "hive.openshift.io/v1",
+					BlockOwnerDeletion: &isTrue,
+					Controller:         &isTrue,
+					Kind:               "ClusterDeployment",
+					Name:               clusterName,
+					UID:                certRequest.GetUID(),
+				}
+
+				certRequest.SetOwnerReferences(append(ownerRefs, ownerRef))
```

Comment:
we should not set the ownerReference within the test like this. We need to check the functionality of the operator/controller.